lib/ansible/modules/system/pamd.py

```diff
@@ -759,6 +836,9 @@ def main():
         changes = service.remove_module_arguments(module.params['type'], module.params['control'], module.params['module_path'],
                                                   module.params['module_arguments'])
     elif action == 'args_present':
+        if [arg for arg in parse_module_arguments(module.params['module_arguments']) if arg.startswith("[")]:
+            module.fail_json(msg="Unable to process bracketed '[' complex arguments with 'args_present'. Please use 'updated'.")
```

Comment:
Couldn't we just use the old behavior of adding it if it didn't exactly match an existing argument, and perhaps emit a warning?

Reply:
That would mean that you would potentially have duplicate arguments on a Pamd line which to me is unexpected and broken behavior. i.e. If I used this module I would not expect it to add duplicate arguments to a rule, in fact if it did that I would consider the module broken...which is why this was changed.

imo `args_present` should really just be used for adding in simple arguments. i.e. `use_authtok` or `sha512` and not for a super complex argument like so:

```
[query=select user_name from internet_service where user_name='%u'  and password=PASSWORD('%p') and service='web_proxy']
```

For those types of arguments one should use updated which is properly able to test equality and update it if necessary.

Reply:
Ok. That's reasonable.